tests/framework/crossplane.go

```diff
@@ -75,6 +74,35 @@ func ValidateNginxFieldExists(conf *Payload, expFieldCfg ExpectedNginxField) err
 	return fmt.Errorf("field not found; expected: %+v\nNGINX conf: %s", expFieldCfg, string(b))
 }
 
+func validateServerBlockDirectives(expFieldCfg ExpectedNginxField, directive Directive) error {
+	for _, serverName := range expFieldCfg.Servers {
+		if directive.Directive == "server" && getServerName(directive.Block) == serverName {
+			for _, serverDirective := range directive.Block {
+				if expFieldCfg.Location == "" && expFieldCfg.fieldFound(serverDirective) {
+					return nil
+				} else if serverDirective.Directive == "location" &&
+					fieldExistsInLocation(serverDirective, expFieldCfg) {
+					return nil
+				}
+			}
+		}
+	}
+	return fmt.Errorf("field not found; expected: %+v\nNGINX conf: %s", expFieldCfg, directive.Directive)
```

Comment:
What if the expFieldCfg doesn't contain Servers? For example, what if we are testing upstream content only? It seems like we would return an error here, which would be incorrect. 

Also, what if the directive isn't a "server" directive? Won't we mistakenly return an error in that case, too?